solutions/fully-configurable/main.tf

```diff
@@ -6,10 +6,23 @@ locals {
 # Resource Group
 ##############################################################################
 
+# Use a local variable to determine if the default resource group should be used
+locals {
+  # Name of default resource group change circa 2019 from default to Default so attempt to return the actual default resource group in this scenario
+  use_default_resource_group = var.existing_resource_group_name == null ? true : lower(var.existing_resource_group_name) == "default"
```

Comment:
I don't understand what you are trying to do here? If `existing_resource_group_name` is `null`, then it should lookup the default resource group using data lookup and use that